src/drivers/frsky_telemetry/sPort_data.c

```diff
@@ -58,31 +58,26 @@
 
 #define frac(f) (f - (int)f)
 
-static int battery_sub = -1;
+//static int battery_sub = -1;
 static int sensor_sub = -1;
 static int global_position_sub = -1;
 static int vehicle_status_sub = -1;
 
+static struct vehicle_status_s vehicle_status;
```

Comment:
This leads to the struct being allocated statically, independent if the application is running or not. So the stack save means that everybody not using sPort still has the hit of having this memory allocated.

You can avoid this by using a pointer using malloc (converting would be fast) and free (only on app exit), or by converting this app into a C++ class and making these member variables.